### ISSUE TITLE
[BUG] Failed consecutive blip check with 0.5.3 and v2_0_0.wav

### ISSUE THREAD

**Issue by bobbob3** [sdatkinson/neural-amp-modeler#236]
Using v0.5.3 pulled shortly before the time of this writing, I am getting an error when using an input/output WAV file set that worked with a previous version of NAM, which was pulled several days ago. The output file was recorded using v2_0_0.wav as the input.

Now I get the following error message when using this input/output file pair that worked previously. I am also getting the error for a new set of input/output files recorded using v2_0_0.wav as the input. However, I am able to train using a different output which was recorded using v1_1_1.wav as the input. I did not try other sets based on v1_1_1.wav to see if they lead to the error message though.

This is the error message:

Delay wasn't provided; attempting to calibrate automatically...
Delays:
 5
 5
After aplying safety factor, final delay is 1
Plotting the delay for manual inspection...
V2 checks...
Replicate ESR is 0.00099014.
Failed consecutive blip check at start of training signal. The target tone doesn't seem to be replicable over short timespans. Is there a noise gate or a time-based effect in the signal chain?
Failed checks; exit training

1. git pull
2. conda update conda
3. conda env update --file environment_gpu.yml --name nam
4. nam
5. Initiate a training from the GUI using the following files as input and output.
[v2_0_0.wav.zip](https://github.com/sdatkinson/neural-amp-modeler/files/11417732/v2_0_0.wav.zip)
[output.wav.zip](https://github.com/sdatkinson/neural-amp-modeler/files/11417729/output.wav.zip)

Expected behavior:
This set of files did not lead to a blip test failure for a past version of NAM.

Desktop:
 - OS: Windows 11
 - Running NAM locally
 - NAM v0.5.3 pulled around 0400 UTC on May 8, 2023.

**Comment by sdatkinson** [comment]
> Replicate ESR is 0.00099014

That's pretty noisy--the data has an ESR of almost 0.001 with itself.  Like the error says-- Is there a noise gate or a time-based effect in the signal chain? or is your recording setup just super-noisy?

I'm still tuning this, so if you wanted to shoot me your reamp & tell me about it I'd love to have a look and keep tweaking things.

**Comment by bobbob3** [comment]
Thank you, and yes the noise was indeed high. I hadn't examined the output file's noise, given that it worked up until now. I was able to improve the noise issue through ground lift settings on the DIs involved and a EBTech hum eliminator, and it looks much cleaner. For some frequencies the noise is now almost 40dB lower.

There were no noise gates or time delays in the signal path., and there are none in these new recordings.

It worked for one new recording, but not for another. They are attached here. The only difference between them is the load box that was used for the amp. The DIs all remain the same. I used outboard DIs and did not use those in the load boxes, so only the load is different. And the loads have quite similar impedance curves. Load box B boosts the highs slightly more than Load box A. They are very similar for low to mid frequencies. Looking at the output files, the noise is only a few dB higher for the output that doesn't work with v0.5.3.

These are the files:
[output2-OK.wav.zip](https://github.com/sdatkinson/neural-amp-modeler/files/11425368/output2-OK.wav.zip)
Same signal path as output3, except that output2 uses load box A. This file works with v0.5.3.

[output3-NG.wav.zip](https://github.com/sdatkinson/neural-amp-modeler/files/11425373/output3-NG.wav.zip)
Same signal path as output2, except that output3 uses load box B. This file gives the following error with v0.5.3.

Delay wasn't provided; attempting to calibrate automatically...
Delays:
 5
 5
After aplying safety factor, final delay is 1
Plotting the delay for manual inspection...
V2 checks...
Replicate ESR is 0.00008851.
Failed start-to-end blip check for blip replicate 1. The target tone doesn't seem to be same at the end of the reamp as it was at the start. Did some setting change during reamping?
Failed checks; exit training

**Comment by sdatkinson** [comment]
Welp, at least it checks out--the ESR between the blip at the start and the end is 0.00107 -- 0.00007 higher than the threshold for the error, 0.001!

I guess I can just back off [the threshold?](https://github.com/sdatkinson/neural-amp-modeler/blob/bc6f76d80f3c06212b7e3661f0997b3b4f4b9c59/nam/train/core.py#L313). If you make this higher, it won't complain.

I'll probably want to make some option to "ignore checks" so that if you really want to insist on going ahead, you can (otherwise I'll probably never hear the end of it from people and plenty of people trying to share how to hack the code to get around it)...

But the intent is to warn that your data are bad, so your model will probably be bad as well (garbage in, garbage out).

I'll keep tweaking it.

**Comment by bobbob3** [comment]
That's indeed close to the threshold! Is that comparing the blip at the start and the end? If so, the main area of difference between the start and end blips in the problem output file looks to exist below 60Hz. There is a slightly longer section of "ringing" below 60Hz for the first blip than the last blip in that file. Otherwise they at least appear similar in a spectrogram for other frequency regions. Perhaps a highpass filter on the blips before that analysis is done? Given that at least for guitar the region doesn't matter much.

Or back off the threshold. This is a high gain capture. Maybe set a different threshold if the user chooses High Gain in the metadata section? I'm assuming this blip difference wouldn't be so pronounced using a cleaner amp setting.

**Comment by sdatkinson** [comment]
Just pushed some updates.

The typical ESRs associated w/ a blip are much larger (ESR is not an absolute! The "0.01" is specific to the test signal that's part of the standard reamp signals I have people use! 😉), so I took some time to calibrate it a bit better.

I also added the ability to ignore it, so that'll be an option (hopefully not needed anymore, but juts in case! 😬)

Should solve your problem, so closing this 🙂
